gobblin-core/src/main/java/gobblin/publisher/TimePartitionedDataWithRowCountsPublisher.java

```diff
@@ -0,0 +1,160 @@
+        // WRITER_OUTPUT_DIR and WRITER_FILE_PATH
+        Path writerOutputDir = WriterUtils.getWriterOutputDir(workUnitState, this.numBranches, branchId);
+
+        if (!this.fss.get(branchId).exists(writerOutputDir)) {
+          LOG.warn("WorkUnit " + workUnitState.getId() + " produced no data");
+          workUnitState.setWorkingState(WorkUnitState.WorkingState.COMMITTED);
+          break;
+        }
+
+        String fileExtension = WriterOutputFormat.valueOf(workUnitState.getProp(ForkOperatorUtils
+                .getPropertyNameForBranch(ConfigurationKeys.WRITER_OUTPUT_FORMAT_KEY, this.numBranches, branchId),
+            WriterOutputFormat.AVRO.name())).getExtension();
+
+        String writerOutputFileName = WriterUtils.getWriterFileName(workUnitState, this.numBranches,
+            branchId, workUnitState.getId(), fileExtension);
+
+        Path writerOutputFullGlob =
+            getWriterOutputGlobMatcher(workUnitState, branchId, writerOutputDir, writerOutputFileName);
+
+        Path writerOutputFilePath = null;
+        for (FileStatus status : this.fss.get(branchId).globStatus(writerOutputFullGlob)) {
+          if (writerOutputFilePath != null) {
+            LOG.warn("Found more than one output for WorkUnit " + workUnitState.getId() + " branch " + branchId
+                + " matching glob: " + writerOutputFullGlob);
+          } else {
+            writerOutputFilePath = status.getPath();
+          }
+        }
+        if (writerOutputFilePath == null) {
+          LOG.error("Unable to find output for WorkUnit " + workUnitState.getId() + " branch " + branchId
+              + "; expected to find file matching glob: " + writerOutputFullGlob);
+          continue;
+        }
+
+        // The directory where the final output directory for this job will be placed. It is a combination of
+        // DATA_PUBLISHER_FINAL_DIR and WRITER_FILE_PATH
+        Path publisherOutputDir = WriterUtils.getDataPublisherFinalDir(workUnitState, this.numBranches, branchId);
+
+        String writerOutputPathStr = writerOutputFilePath.getParent().toString();
+
+        // The suffix at the end of the writer output path, e.g. 2015/06/29/10
+        String pathSuffix = writerOutputPathStr.substring(
+            writerOutputPathStr.indexOf(writerOutputDir.toString()) + writerOutputDir.toString().length() + 1);
+
+        int recordsWritten = workUnitState.getPropAsInt(ForkOperatorUtils
+            .getPropertyNameForBranch(ConfigurationKeys.WRITER_ROWS_WRITTEN, this.numBranches, branchId));
+
+        // Remove extension, insert number of records written, put extension back on the end
+        String outputFileName = writerOutputFileName.substring(0,
+            writerOutputFileName.length() - fileExtension.length()) + recordsWritten + "." + fileExtension;
+
+        Path outputPath = new Path(publisherOutputDir, pathSuffix + Path.SEPARATOR + outputFileName);
```

Comment:
This method is too long. You can separate some parts into smaller methods such as `getOutputPath()` etc.